astropy/units/quantity.py

```diff
@@ -1041,6 +1083,9 @@ def quantity_iter():
         return quantity_iter()
 
     def __getitem__(self, key):
+        if isinstance(key, str) and isinstance(self.unit, StructuredUnit):
+            return self._new_view(self.view(np.ndarray)[key], self.unit[key])
```

Comment:
So this is going through a view of the value, selecting one of the fields.
How is ``self.view(np.ndarray)[key]`` different from ``.value[key]``?

Reply:
It's the same except just a tiny bit faster.